src/fmu/dataio/providers/_fmu.py

```diff
@@ -84,318 +97,239 @@ class FmuProvider:
-            mreal["name"] = self._real_name
-            mreal["parameters"] = self._ert_info["params"]
-
-            if self.include_ertjobs:
-                mreal["jobs"] = self._ert_info["jobs"]
+            return None
+
+        restart_metadata = internal.CaseSchema.model_validate(
+            ut.yaml_load(restart_case_metafile, loader="standard")
+        )
+        return _utils.uuid_from_string(
+            f"{restart_metadata.fmu.case.uuid}{restart_path.name}"
+        )
+
+    def _get_ert_parameters(self) -> meta.Parameters | None:
+        logger.debug("Read ERT parameters")
+        assert self._runpath is not None
+        parameters_file = self._runpath / "parameters.txt"
+        if not parameters_file.exists():
+            warn("The parameters.txt file was not found", UserWarning)
+            return None
+
+        params = _utils.read_parameters_txt(parameters_file)
+        logger.debug("parameters.txt parsed.")
+        # BUG(?): value can contain Nones, loop in fn. below
+        # does contains check, will fail.
+        return meta.Parameters(root=_utils.nested_parameters_dict(params))  # type: ignore
+
+    def _get_ert_jobs(self) -> dict | None:
+        logger.debug("Read ERT jobs")
+        assert self._runpath is not None
+        jobs_file = self._runpath / "jobs.json"
+        if not jobs_file.exists():
+            logger.debug("jobs.json was not found")
+            return None
+
+        with open(jobs_file) as stream:
+            logger.debug("parsing jobs.json.")
+            return json.load(stream)
+
+    def _get_iteration_and_real_uuid(self, case_uuid: UUID) -> tuple[UUID, UUID]:
+        iter_uuid = _utils.uuid_from_string(f"{case_uuid}{self._iter_name}")
+        real_uuid = _utils.uuid_from_string(f"{case_uuid}{iter_uuid}{self._real_id}")
+        return iter_uuid, real_uuid
+
+    def _get_fmucase_meta(self) -> meta.FMUCase:
+        """Parse and validate the CASE metadata."""
+        logger.debug("Loading case metadata file and return pydantic case model")
+        assert self._casepath is not None
+        case_metafile = self._casepath / ERT_RELATIVE_CASE_METADATA_FILE
+        case_meta = internal.CaseSchema.model_validate(
+            ut.yaml_load(case_metafile, loader="standard")
+        )
+        return case_meta.fmu.case
+
+    def _get_realization_meta(self, real_uuid: UUID) -> meta.Realization:
+        return meta.Realization(
+            id=self._real_id,
+            name=self._real_name,
+            parameters=self._get_ert_parameters(),
+            jobs=self._get_ert_jobs() if self.include_ertjobs else None,
+            uuid=real_uuid,
+        )
+
+    def _get_iteration_meta(self, iter_uuid: UUID) -> meta.Iteration:
+        return meta.Iteration(
+            id=self._iter_id,
+            name=self._iter_name,
+            uuid=iter_uuid,
+            restart_from=self._get_restart_data_uuid()
+            if environ.get(RESTART_PATH_ENVNAME)
+            else None,
```

Comment:
```suggestion
            restart_from=(
                self._get_restart_data_uuid()
                if environ.get(RESTART_PATH_ENVNAME)
                else None
            ),
```
I think this adds a lot to the readability